Steps and Boilerplate (#2)
vtex-apps/service-course: course/steps/01-boilerplate/en.md

> In the `manifest.json` file, you will find the app's name, vendor, version, and other informations to pay attention to: builders, policies and dependencies. In this initial state, we have the following configurations:

Reviewer: ```suggestion
In the `manifest.json` file, you will find the app's name, vendor, version, and other information to pay attention to: builders, policies and dependencies. In this initial state, we have the following configurations:
```

> - builders: what builder your app will need. In this case, we have, so far, only the `docs builder` and the `node builder`, with its respectives versions;

Reviewer 2: ```suggestion
- builders: what builders your app will need. In this case, we have, so far, only the `docs builder` and the `node builder`, with its respective versions;
```

> - builders: what builder your app will need. In this case, we have, so far, only the `docs builder` and the `node builder`, with its respectives versions;

Reviewer: ```suggestion
- builders: what builder your app will need. In this case, we have, so far, only the `docs builder` and the `node builder`, with their respective versions;
```

> - policies: if the app being built needs to access some external services or get some specific data from other places, it needs to declare so, even for external API’s. At this point, we have no specific policies yet;

Reviewer: ```suggestion
- policies: if the app being built needs to access some external services or get some specific data from other places, it needs to declare so, even for external APIs. At this point, we have no specific policies yet;
```

- `/node/clients`: both files are almost blank and are now just placeholders for the next steps.

- `/node/handlers`: contains a handler that will be used in the next steps.

- `/node/utils`: you will find a file containing global constants declarations (`/node/constants.ts`).

> - `/node/index.ts`: contains the initial declarations for the app functionality like the cache declaration and the service declarations, which will be incremented during the course. Here is also possible to export resolver functions.

Reviewer: ```suggestion
- `/node/index.ts`: contains the initial declarations for the app functionality like the cache declaration and the service declarations, which will be incremented during the course. Here is also possible to export resolver functions implementations.
```

- `/node/service.json`: It describes your REST API and some characteristics that will directly impact your app's infrastructure attributes.
  Your service.json file will be found inside your app's /node folder, and will look similar to this:

  ```
  {
  "memory": 256,
  "timeout": 2,
  "minReplicas": 2,
  "maxReplicas": 4,
  "routes": {
    "status": {
      "path": "/_v/status/:code",
      "public": true
    }
  }
  ```

>   - memory: In MegaBytes. How much memory your app will have allocated. This value will be overwritten if IO detects that your app is abusing memory usage.

Reviewer 3: Hmm, vi que já tá aqui... Então acho que aqui poderia ser a tabela. Não precisa o exemplo ter tudo, mas coloca as outras coisas que dá pra fazer.

"timeout": 2,
  "minReplicas": 2,
  "maxReplicas": 4,
podem ser omitidos sem problema, se não me engano.

Reviewer: Acho que uma tabelinha seria legal de fato 🤔 
| field       | description              | optional  |
|---------|--------------------|----------|
| timeout | configures timeout | **false** |

  - timeout: In seconds. VTEX.IO infra will abort the connection if the request time is longer than timeout
  - minReplicas: When your app is running, how many minimum replicas will be available.
  - maxReplicas: The largest amount of replicas that will be available.
  - routes: Describes your app's REST routes, inside you will descibe the name, (ex: ssr), the path, and if its public or private

## `/graphql` Directory Overview

> 

Reviewer: aqui embaixo (não consigo comentar :( ) tem um `provinding` -> `providing`

Reviewer: ```diff
- clone 
+ clone (`git clone`)
```
